narwhals/_duckdb/utils.py

```diff
@@ -18,7 +20,13 @@
     from narwhals.utils import Version
 
 
-def maybe_evaluate(df: DuckDBLazyFrame, obj: Any) -> Any:
+class ExprKind(Enum):
+    LITERAL = auto()  # e.g. nw.lit(1)
```

Comment:
Might be worth adding a tiny docstring and describe interaction between values:
```suggestion
class ExprKind(Enum):
    """Describe which kind of expression we are dealing with.
    
    Composition rule is:
    - LITERAL & LITERAL -> LITERAL
    - TRANSFORM & X -> TRANSFORM 
    - X & TRANSFORM -> TRANSFORM
    - all remaining cases -> AGGREGATION
    """
    LITERAL = auto()  # e.g. nw.lit(1)
```